patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0001-fix-for-source-build.patch

```diff
@@ -37,71 +36,87 @@ index 7db18927..1fc72f5b 100644
--    <MicrosoftCSharpVersion>4.5.0</MicrosoftCSharpVersion>
-+    <MicrosoftCSharpVersion>4.7.0</MicrosoftCSharpVersion>
-     <MicrosoftSourceLinkGitHubVersion>1.0.0</MicrosoftSourceLinkGitHubVersion>
-     <NetStandardVersion>2.0.3</NetStandardVersion>
-     <SystemCollectionsSpecializedVersion>4.3.0</SystemCollectionsSpecializedVersion>
-@@ -15,7 +15,7 @@
-     <SystemRuntimeSerializationPrimitivesVersion>4.3.0</SystemRuntimeSerializationPrimitivesVersion>
+@@ -14,8 +14,8 @@
      <SystemRuntimeSerializationXmlVersion>4.3.0</SystemRuntimeSerializationXmlVersion>
      <SystemSecurityClaimsVersion>4.3.0</SystemSecurityClaimsVersion>
+     <SystemTextEncoding>4.3.0</SystemTextEncoding>
+-    <SystemTextJson>4.7.2</SystemTextJson>
+-    <SystemTextEncodingsWeb>4.7.2</SystemTextEncodingsWeb>
++    <SystemTextJson>7.0.0</SystemTextJson>
++    <SystemTextEncodingsWeb>7.0.0</SystemTextEncodingsWeb>
+     <SystemXmlXmlDocumentVersion>4.3.0</SystemXmlXmlDocumentVersion>
+   </PropertyGroup>
+ 
+@@ -25,8 +25,8 @@
+   </PropertyGroup>
+ 
+   <PropertyGroup Condition="'$(TargetFramework)' != 'net8.0'">
 -    <SystemSecurityCryptographyCngVersion>4.5.0</SystemSecurityCryptographyCngVersion>
+-    <MicrosoftCSharpVersion>4.5.0</MicrosoftCSharpVersion>
 +    <SystemSecurityCryptographyCngVersion>5.0.0</SystemSecurityCryptographyCngVersion>
-     <SystemXmlXmlDocumentVersion>4.3.0</SystemXmlXmlDocumentVersion>
++    <MicrosoftCSharpVersion>4.7.0</MicrosoftCSharpVersion>
+     <MicrosoftSourceLinkGitHubVersion>1.0.0</MicrosoftSourceLinkGitHubVersion>
    </PropertyGroup>
  </Project>
 diff --git a/build/targets.props b/build/targets.props
-index e5ac9a6e..62004f28 100644
+index f020b59b..934aae74 100644
 --- a/build/targets.props
 +++ b/build/targets.props
-@@ -1,6 +1,6 @@
+@@ -1,7 +1,7 @@
  <Project>
    <PropertyGroup>
--    <SrcTargets>net45;net461;net472;netstandard2.0;net6.0</SrcTargets>
-+    <SrcTargets>netstandard2.0;net6.0</SrcTargets>
+-    <SrcTargets Condition="'$(TargetNet8)' == 'True'">net461;net462;net472;netstandard2.0;net6.0;net8.0</SrcTargets>
+-    <SrcTargets Condition="'$(TargetNet8)' != 'True'">net461;net462;net472;netstandard2.0;net6.0</SrcTargets>
++    <SrcTargets Condition="'$(TargetNet8)' == 'True'">netstandard2.0;net6.0;net8.0</SrcTargets>
++    <SrcTargets Condition="'$(TargetNet8)' != 'True'">netstandard2.0;net6.0</SrcTargets>
      <SrcStandardTargets>netstandard2.0</SrcStandardTargets>
    </PropertyGroup>
  </Project>
 diff --git a/build/targetsTest.props b/build/targetsTest.props
-index 19306fa5..0f430eb8 100644
+index a635db69..35910de1 100644
 --- a/build/targetsTest.props
 +++ b/build/targetsTest.props
-@@ -1,6 +1,6 @@
+@@ -1,7 +1,7 @@
  <Project>
    <PropertyGroup>
--    <TestTargets>net452;net461;net472;netcoreapp2.1;net6.0</TestTargets>
-+    <TestTargets>netcoreapp2.1;net6.0</TestTargets>
+-    <TestTargets Condition="'$(TargetNet8)' == 'True'">net461;net462;net472;netcoreapp2.1;net6.0;net8.0</TestTargets>
```

Comment:
Are we building the tests during source-build? It would be nice if we could avoid that.

Reply:
We only build the projects that are outlined [here](https://github.com/dotnet/source-build-externals/blob/a10b4bacb0f29c64eed40cefdeb6fbe0dc82eb01/repo-projects/azure-activedirectory-identitymodel-extensions-for-dotnet.proj#L27-L60).

Technically, this change isn't necessary but I used find/replace to remove all the references to the netfx target framework.